lib/init.lua

```diff
@@ -326,28 +339,30 @@ local function archetypeTraverseRemove(world: World, componentId: i53, archetype
 	local from = (archetype or world.ROOT_ARCHETYPE) :: Archetype
 	local edge = ensureEdge(from, componentId)
 
-
-	if not edge.remove then
-		local to = table.clone(from.types) 
+	local remove = edge.remove
+	if not remove then
+		local to = table.clone(from.types)
 		table.remove(to, table.find(to, componentId))
-		edge.remove = ensureArchetype(world, to, from)
+		remove = ensureArchetype(world, to, from)
+		edge.remove = remove :: never
 	end
 
-	return edge.remove
+	return remove
 end
 
-function World.remove(world: World, entityId: i53, componentId: i53) 
-	local record = ensureRecord(world.entityIndex, entityId)
+function World.remove(world: World, entityId: i53, componentId: i53)
+	local entityIndex = world.entityIndex
+	local record = ensureRecord(entityIndex, entityId)
 	local sourceArchetype = record.archetype
 	local destinationArchetype = archetypeTraverseRemove(world, componentId, sourceArchetype)
 
-	if sourceArchetype and not (sourceArchetype == destinationArchetype) then 
-		moveEntity(world.entityIndex, entityId, record, destinationArchetype)
+	if sourceArchetype and not (sourceArchetype == destinationArchetype) then
+		moveEntity(entityIndex, entityId, record, destinationArchetype)
 	end
 end
 
 -- Keeping the function as small as possible to enable inlining
-local function get(componentIndex: { [i24]: ArchetypeMap }, record: Record, componentId: i24)
+local function get(_componentIndex: {[i24]: ArchetypeMap}, record: Record, componentId: i24)
```

Comment:
We should just remove componentIndex from the parameters to be honest. It was kept before but it is now obsolete with archetype.records